src/to_java.rs

```diff
@@ -206,15 +205,16 @@ where
     }
 }
 
-impl<J> ToJavaImpl<J> for &J
+impl<J, R> ToJavaImpl<J> for &R
 where
     J: Upcast<java::lang::Object>,
+    R: ?Sized + ToJavaImpl<J>
 {
     fn to_java_impl<'jvm>(
         rust: &Self,
         jvm: &mut Jvm<'jvm>,
     ) -> crate::Result<'jvm, Option<Local<'jvm, J>>> {
-        Ok(Some(jvm.local(rust)))
+        ToJavaImpl::to_java_impl(*rust, jvm)
```

Comment:
```suggestion
        R::to_java_impl(rust, jvm)
```

In general, my preference in these kind of situations is to specify the receiver type (`R`) that we are invoking the method on. It also means we don't have to on `*rust` but can let the compiler manage the derefs.